docs/page-specific-js.md

```diff
@@ -1,26 +1,42 @@
 # How to Add JS for a Page
+
 1. Make JS
 2. Add a Hugo variable
 3. Import script with Hugo variable
 4. ...
 5. Debug!
 
 ## Make JS
+
 You'll need to modify the theme that's applied to the markdown that Hugo processes in order to add new JS to any pages. First, ask yourself if __everyone__ needs to have this or if it can be added in a partial.
 
 ### Ading to a Partial
+
 1. Add your fancy new JS in a file, `fancy.js` in `themes/jb/assets/js`
-2. Go find whichever partial your code belongs to, `fancy.html`
+2. Go find whichever partial your code belongs to, `fancy.html`, and put all of the following in a `{{ define "headcontent" }}` hugo block
 3. Add a Hugo reference to the HTML
+
     ``` go
     {{ $fancy := resources.Get "js/fancy.js" }}
     ```
-4. Add a `script` tag for the processed JS
+
+4. Add an in production block, so the JS can be properly minified via hugo
+
+    ```go
+    {{ if hugo.IsProduction }}
+      {{ $fancy = $fancy | minify | fingerprint }}
+    {{ end }}
+    ```
+
+5. Add a `script` tag for the processed JS
+
     ``` html
-    <script src="{{ $fancy.PermaLink }}"></script>
+    <script type="text/javascript" src="{{ $fancy.PermaLink }}" integrity="{{ $fancy.Data.Integrity }}"></script>
```

Comment:
`type="text/javascript"` is optional and not necessary when you are using the correct mime type to serve JavaScript.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script
```suggestion
    <script src="{{ $fancy.PermaLink }}" integrity="{{ $fancy.Data.Integrity }}"></script>
```

Reply:
I know it's optional, but is there a reason why it's bad to include it?

I've heard of certain attacks (I don't think it applies to this particular circumstance) that if the server is not specifying the right Content-Type and a user can interact with the server (somehow \*\*waving magical hands\*\*) then they can possibly [get xss](https://portswigger.net/kb/issues/00800400_content-type-incorrectly-stated). Again it's probably not super applicable to this situation, definitely not a web expert at all, but does it hurt anything to have it there?

Reply:
I guess it doesn't hurt, besides adding a few more bytes to the page download. 😉